### PR TITLE
Enhancements and Fixes

### LINKED ISSUE #31: Improvements on agreements page
- add a "create new agreement" link on agreements list page
- use a form framework to validate fields (value numeric, wallet address, etc)

### LINKED ISSUE #33: bug when adding a payment to an already paid agreement
https://www.loom.com/share/68d5256bcbd64dca9fe89b14408bea6f?focus_title=1&muted=1&from_recorder=1

### LINKED ISSUE #34: improvements on dashboard
- list only 2 or 3 payments in list, click "more..." to see all payments in the payments page
- bug: duplicating payments - it happens when I am the payer and the receiver. It should not allow user to create agreement for himself or pay himself.
- should be discriminated if payment is

### PR DESCRIPTION
**Pull request by nomadbitcoin**
1. fix: Payment processing error with decimals.
2. fix: Resolved the issue of duplicated payments. Also, added a validation to prevent users from creating agreements for themselves.
3. feat: Updated the payments list to display only the latest 2 payments. Added a "More" button to view all payments.
4. feat: Introduced a "Create New Agreement" link on the agreements list page for better user accessibility.
5. chore: Integrated a form framework for efficient and robust field validation.

Issues Addressed:

Fix #31 : Issue with duplicated payments.
Fix #33 : Enhancements to the payments list display.
Fix #34 : Added a direct link for creating new agreements.

### CODE REVIEW

File: components/AddAgreement/AddAgreement.js [hunk 1]
```diff
@@ -1,56 +1,90 @@
...
   const [isLoading, setIsLoading] = useState(false)
   const [transactionHash, setTransactionHash] = useState(null)
   const { contract, loading } = useAgreementContract()
+  const [formErrors, setFormErrors] = useState({});
   const router = useRouter()
+  const { account } = useAccount();
+
+  const AgreementSchema = Yup.object().shape({
+    title: Yup.string()
+      .required('Title is required'),
```
Comment (danicuki) [inline components/AddAgreement/AddAgreement.js:26]: these texts should not be hard coded but rather be in a translation dictionary so the platform can be easily translated to other languages

**Review by danicuki — APPROVED** [review]
I am approving this PR, but we should create a task to add tests on the front-end for the futures implemented.

**Comment by danicuki** [comment]
@nomadbitcoin conflicts with other PR. You need to rebase to get it accepted

**Comment by nomadbitcoin** [comment]
Corrected, I had accepted Mika's pr after opening my

**Comment by nomadbitcoin** [comment]
Corrected, I had accepted Mika's pr after opening my

**Comment by danicuki** [comment]
You need to open a new PR. This one is closed

**Pull request closed without merging**